src/Datagrid/Pager.php

```diff
@@ -43,22 +47,14 @@ class Pager extends BasePager
      */
     public function computeNbResult()
     {
-        $countQuery = clone $this->getQuery();
-
-        if (\count($this->getParameters()) > 0) {
-            $countQuery->setParameters($this->getParameters());
-        }
-
-        if (\count($this->getCountColumn()) > 1) {
-            $this->countCompositePrimaryKey($countQuery);
-        } else {
-            $this->countSinglePrimaryKey($countQuery);
+        if (!$this->getQuery() instanceof ProxyQueryInterface) {
+            throw new \InvalidArgumentException(
```

Comment:
Since the checked value is not an argument of this method, I think we should use `\TypeError` instead,